src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt

```diff
@@ -56,4 +61,33 @@ class CertificateService(
         return certificateMapping.toCertificateDocumentResponseDto(result)
 
     }
+
+    fun getCertificatesByBpn(bpn: String, pageRequest: PageRequest): PageDto<CertificateDocumentResponseDto> {
+        if (bpn.isBlank())
+            throw IllegalArgumentException("Provided business partner number is null or empty")
+
+        return when {
+            bpn.startsWith("BPNL") -> {
+                val certificates = certificateRepository.findByBusinessPartnerNumber(bpn, pageRequest)
+                if (certificates.isEmpty) {
+                    throw CertificateNotExists("Legal Entity", bpn)
```

Comment:
These checks for empty page leads to a bit of a strange behaviour in the API. When I paginate to a page above the limit I then get this certificate does not exist error. However, the certificates   for this BPN actually exist in the previous pages.
I would rather expect to just get an empty page or at least get a matching error message that I paginated over the total element collection.

Reply:
Thanks for detailed explanation, i have adapted changes in same pull request.